client/tools/settings.coffee

```diff
@@ -81,3 +85,44 @@ defineTool
   click: ->
     return unless (room = currentRoom.get())?
     room.gridSnap.set not room.gridSnap.get()
+
+defineTool
+  name: 'Transparency'
+  category: 'color'
+  icon: 'highlighter'
+  help: 'Change transparency of pen'
+  active: ->
+    allowTransparency.get()
+  click: ->
+    allowTransparency.set not allowTransparency.get()
+    updateOpacity 1.0, allowTransparency.curValue
+  init: ->
+    updateOpacity 1.0, false
+
+
+# These values are chosen for no particular reason.  I saw that
+# 12.5 was a number you liked for highlighting Perhaps .25 should be 12.5
+for opacity in [.75, .50, .25]
+  do (opacity) ->
+    defineTool
+      name: "Opacity:#{opacity*100}"
+      category: 'color'
+      icon: "opacity#{opacity*100}"
+      help: "Select #{opacity*100}% Transparency"
+      click: ->
+        updateOpacity opacity, true
+      active: ->
+        if currentOpacity.get() == opacity then true else false
+
+
+updateOpacity = (val, display) ->
+  allowTransparency.set display
+
+  if currentOpacity.get() == val
+    currentOpacity.set 1.0
```

Comment:
I don't like resetting `currentOpacity` to 1 when toggling the mode off. I'd rather keep the same opacity when I leave and re-enter transparent mode. Similar to how fill mode works, we should have a Boolean `currentOpacityOn`, which should replace the current `allowTransparency`.

Similarly, I don't like that clicking on the transparent mode for the first time doesn't immediately turn on transparency. It (and thus `currentOpacity`) should start at a global default (maybe 50%?), so one of the specific buttons starts down as well.

Reply:
bye bye allowTransparency. √